packages/schema-migrations/src/modifications/annotations/GenericAnnotationModification.ts

```diff
@@ -0,0 +1,130 @@
+import { MigrationBuilder } from '@contember/database-migrations'
+import { Schema } from '@contember/schema'
+import { SchemaUpdater, updateModel, updateEntity, updateField } from '../utils/schemaUpdateUtils'
+import { createModificationType, Differ, ModificationHandler } from '../ModificationHandler'
+import { Migration } from '../../Migration'
+
+export interface BaseAnnotationModificationData {
+	entityName: string
+	fieldName?: string
+}
+
+export type AnnotationModificationData<T extends string> = BaseAnnotationModificationData & {
+	[K in T]?: string
+}
+
+export interface AnnotationConfig<T extends string> {
+	id: string
+	annotationField: T
+	displayName: string
+}
+
+export function createAnnotationModification<T extends string>(config: AnnotationConfig<T>) {
```

Comment:
I would make the generic param more strict, maybe:


```suggestion
import { MigrationBuilder } from '@contember/database-migrations'
import { Model, Schema } from '@contember/schema'
import { SchemaUpdater, updateModel, updateEntity, updateField } from '../utils/schemaUpdateUtils'
import { createModificationType, Differ, ModificationHandler } from '../ModificationHandler'
import { Migration } from '../../Migration'

export interface BaseAnnotationModificationData {
	entityName: string
	fieldName?: string
}

export type AnnotationField = keyof Model.AnyField & keyof Model.Entity

export type AnnotationModificationData<T extends AnnotationField> = BaseAnnotationModificationData & {
	[K in T]?: string
}


export interface AnnotationConfig<T extends AnnotationField> {
	id: string
	annotationField: T
	displayName: string
}

export function createAnnotationModification<T extends AnnotationField>(config: AnnotationConfig<T>) {
```